File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.py
def solution(a, b):
    answer = 0
    if a * b % 2 != 0:
        answer = a ** 2 + b ** 2
    else:
        if a % 2 == 1 or b % 2 == 1:
            answer = 2 * (a + b)
        else:
            answer = abs(a-b)
        
    return answer

File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.java
class Solution {
    public int solution(int a, int b) {
        int answer = 0;
        
        int a_odd = a % 2;
        int b_odd = b % 2;
        
        if (a_odd == 1 && b_odd == 1) {
            answer = a * a + b * b;
        } else {
            if (a_odd == 0 && b_odd == 0) {
                answer = Math.abs(a - b);
            } else {
                answer = 2 * (a + b);
            }
        }
        
        return answer;
    }
}